dissect/target/helpers/configutil.py

```diff
@@ -164,31 +183,112 @@ def parse_file(self, fh: TextIO) -> None:
             prev_key, *value = self.SEPERATOR.split(line, 1)
             value = value[0].strip() if value else ""
 
-            if old_value := information_dict.get(prev_key):
-                if not isinstance(old_value, list):
-                    old_value = [old_value]
-                value = old_value + [value]
-
-            information_dict[prev_key] = value
+            _update_dictionary(information_dict, prev_key, value)
 
         self.parsed_data = information_dict
 
-    def _parse_line(self, line: str, data_dict: dict) -> tuple:
-        key, *values = self.SEPERATOR.split(line)
-        values = " ".join(value for value in values if value).strip()
 
-        if old_value := data_dict.get(key):
-            if not isinstance(old_value, list):
-                old_value = [old_value]
-            values = old_value + [values]
-        data_dict[key] = values
+class PeekableIterator:
+    """Source gotten from:
+    https://more-itertools.readthedocs.io/en/stable/_modules/more_itertools/more.html#peekable
+    """
+
+    def __init__(self, iterable):
+        self._iterator = iter(iterable)
+        self._cache = deque()
+
+    def __iter__(self):
+        return self
+
+    def peek(self):
+        if not self._cache:
+            try:
+                self._cache.append(next(self._iterator))
+            except StopIteration:
+                return
+        return self._cache[0]
+
+    def __next__(self):
+        if self._cache:
+            return self._cache.popleft()
+
+        return next(self._iterator)
+
+
+class Indentation(Default):
```

Comment:
an you add a docstring indicating the type of config it is supposed to read? Proabably also good to add to the other config classes.